Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -149,16 +215,38 @@ - (void)toxAV:(OCTToxAV *)toxAV receiveCallAudioEnabled:(BOOL)audio videoEnabled
     OCTCall *call = [[OCTCall alloc] initCallWithChat:chat];
     call.status = OCTCallStatusIncoming;
 
-    [self.mutableCalls addObject:call];
+    [self.calls addCall:call];
 
-    [self.delegate callSubmanager:self receiveCall:call audioEnabled:audio videoEnabled:video];
+    if ([self.delegate respondsToSelector:@selector(callSubmanager:receiveCall:audioEnabled:videoEnabled:)]) {
+        [self.delegate callSubmanager:self receiveCall:call audioEnabled:audio videoEnabled:video];
+    }
 }
 
 - (void)toxAV:(OCTToxAV *)toxAV callStateChanged:(OCTToxAVCallState)state friendNumber:(OCTToxFriendNumber)friendNumber
-{}
+{
+    OCTDBManager *dbManager = [self.dataSource managerGetDBManager];
+    OCTDBChat *dbChat = [dbManager getOrCreateChatWithFriendNumber:friendNumber];
+    OCTChat *chat = [self.chatConverter objectFromRLMObject:dbChat];
+
+    [self.calls updateCallWithChat:chat updateBlock:^(OCTCall *call) {
+        call.state = state;
+        if (((state & OCTToxAVCallStateError) != 0) || ((state & OCTToxAVCallStateFinished) != 0)) {
+            call.status = OCTCallStatusInactive;
+        }
+    }];
+}
 
 - (void)toxAV:(OCTToxAV *)toxAV audioBitRateChanged:(OCTToxAVAudioBitRate)bitrate stable:(BOOL)stable friendNumber:(OCTToxFriendNumber)friendNumber
-{}
+{
+    if ([self.delegate respondsToSelector:@selector(callSubmanager:audioBitRateChanged:stable:forCall:)]) {
+        OCTDBManager *dbManager = [self.dataSource managerGetDBManager];
+        OCTDBChat *dbChat = [dbManager getOrCreateChatWithFriendNumber:friendNumber];
+        OCTChat *chat = [self.chatConverter objectFromRLMObject:dbChat];
+
+        OCTCall *call = [self.calls callWithFriend:chat.friends.firstObject];
```

Comment:
You can create helper method in this class (`OCTSubmanagerCalls`) to create call from friendNumber and use it in all delegate methods